packages/ui/src/components/Visualization/ContextToolbar/ExportDocument/EntitiesList.tsx

```diff
@@ -0,0 +1,93 @@
+import { Divider, Icon, MenuItemAction, SelectGroup, SelectList, SelectOption } from '@patternfly/react-core';
+import { EyeIcon, EyeSlashIcon } from '@patternfly/react-icons';
+import { FunctionComponent } from 'react';
+import { BaseCamelEntity } from '../../../../models/camel/entities';
+import { DocumentationEntity } from '../../../../models/documentation';
+
+export type IEntityVisibility = {
+  entity: BaseCamelEntity;
+  isVisible: boolean;
+};
+
+type IEntitiesList = {
+  documentationEntities: DocumentationEntity[];
+  onToggleVisibility: (index: number) => void;
+  onToggleAll: () => void;
+};
+
+export const EntitiesList: FunctionComponent<IEntitiesList> = ({
+  documentationEntities,
+  onToggleVisibility,
+  onToggleAll,
+}) => {
+  const allEntitiesVisible = !documentationEntities.find((e) => !e.isVisible);
+
+  type IEntityOption = {
+    entityLabel: string;
+    isVisible: boolean;
+    entityIndex?: number;
+    onToggleVisibility: (index?: number) => void;
+  };
+
+  const EntityOption: FunctionComponent<IEntityOption> = ({
```

Comment:
Creating components inside of components it's harmful, could we extract this `EntityOption` to be a standalone component instead?

Reply:
And probably taking the `IEntityOption` type with it.

Reply:
done :+1: